app/models/reports/repository.rb

```diff
@@ -16,112 +23,55 @@ def initialize(regions, periods:, reporting_schema_v2: false)
-    attr_reader :control_rate_query_v2
-    attr_reader :earliest_patient_data_query
     attr_reader :follow_ups_query
     attr_reader :no_bp_measure_query
     attr_reader :period_type
     attr_reader :periods
     attr_reader :regions
     attr_reader :registered_patients_query
+    attr_reader :schema
 
     def reporting_schema_v2?
       @reporting_schema_v2
     end
 
     delegate :cache, :logger, to: Rails
 
-    # Returns the earliest patient record for a Region from either assigned or registered patients. Note that this *ignores*
-    # the periods that are passed in for the Repository - this is the true 'earliest report date' for a Region.
-    memoize def earliest_patient_recorded_at
-      region_entries = regions.map { |region| RegionEntry.new(region, __method__) }
-      cached_results = cache.fetch_multi(*region_entries, force: bust_cache?) { |region_entry|
-        earliest_patient_data_query.call(region_entry.region)
-      }
-      cached_results.each_with_object({}) { |(region_entry, time), results| results[region_entry.slug] = time }
-    end
-
-    memoize def earliest_patient_recorded_at_period
-      earliest_patient_recorded_at.each_with_object({}) { |(slug, time), hsh| hsh[slug] = Period.new(value: time, type: @period_type) if time }
-    end
-
-    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
-    # are the basis for control rates. These counts DO include lost to follow up.
-    memoize def adjusted_patients_with_ltfu
-      cumulative_assigned_patients.each_with_object({}) do |(entry, result), results|
-        values = periods.each_with_object(Hash.new(0)) { |period, region_result|
-          region_result[period] = result[period.adjusted_period]
-        }
-        results[entry] = values
-      end
-    end
-
-    # Adjusted patient counts are the patient counts from three months ago (the adjusted period) that
-    # are the basis for control rates. These counts DO NOT include lost to follow up.
-    memoize def adjusted_patients_without_ltfu
-      cumulative_assigned_patients.each_with_object({}) do |(entry, result), results|
-        values = periods.each_with_object(Hash.new(0)) { |period, region_result|
-          next unless result.key?(period.adjusted_period)
-          region_result[period] = result[period.adjusted_period] - ltfu[entry][period]
-        }
-        results[entry] = values
-      end
-    end
+    DELEGATED_METHODS = [
+      :adjusted_patients_with_ltfu,
+      :adjusted_patients_without_ltfu,
+      :assigned_patients,
+      :complete_monthly_registrations,
+      :controlled_rates,
+      :controlled,
+      :cumulative_assigned_patients,
+      :cumulative_registrations,
+      :earliest_patient_recorded_at,
+      :earliest_patient_recorded_at_period,
+      :ltfu_rates,
+      :ltfu,
+      :monthly_registrations,
+      :uncontrolled_rates,
+      :uncontrolled
+    ]
+
+    delegate(*DELEGATED_METHODS, to: :schema)
```

Comment:
All of these methods are implemented in both v1 and v2 schemas, hence the delegation.

Everything else below is either not retrievable yet from v2 (follows up by user), or is something that I haven't dug into yet in terms of how to grab it from the v2 schema.